ecommerce/views.py

```diff
@@ -285,3 +385,20 @@ def _update_coupons(basket, product_version, coupons):
                 product_version.product, basket.user, auto_only=True
             )
     return coupon_version
+
+
+def coupon_code_csv_view(request, version_id):
+    """View for returning a csv file of coupon codes"""
+    if not (request.user and request.user.is_staff):
```

Comment:
There's a `django-contrib` `@staff_member_required` decorator which is worth using here: https://github.com/django/django/blob/master/django/contrib/admin/views/decorators.py#L5

Reply:
Using `@staff_member_required` will redirect to `/admin/login/?next=/couponcodes/<id>` even if already logged in, I think it might be better to just raise a 403.